.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,104 @@
+name: Unit test for Pyodide
+
+on:
+  schedule:
+    # Nightly build at 3:42 A.M.
+    - cron: "42 3 */1 * *"
+  push:
+    branches:
+      - main
+      # Release branches
+      - "[0-9]+.[0-9]+.X"
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+.X"
+  # Manual run
+  workflow_dispatch:
+
+env:
+  FORCE_COLOR: 3
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  check_build_trigger:
+    name: Check build trigger
+    runs-on: ubuntu-latest
+    if: github.repository == 'scikit-learn/scikit-learn'
+    outputs:
+      build: ${{ steps.check_build_trigger.outputs.build }}
+    steps:
+      - name: Checkout scikit-learn
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+
+      - id: check_build_trigger
+        name: Check build trigger
+        shell: bash
+        run: |
+          set -e
+          set -x
+
+          COMMIT_MSG=$(git log --no-merges -1 --oneline)
+
+          # The commit marker "[pyodide]" will trigger the build when required
+          if [[ "$GITHUB_EVENT_NAME" == schedule ||
+                "$GITHUB_EVENT_NAME" == workflow_dispatch ||
+                "$COMMIT_MSG" =~ \[pyodide\] ]]; then
+              echo "build=true" >> $GITHUB_OUTPUT
+          fi
+
+  build_wasm_wheel:
+    name: Build WASM wheel
+    runs-on: ubuntu-latest
+    needs: check_build_trigger
+    if: needs.check_build_trigger.outputs.build
+    steps:
+      - name: Checkout scikit-learn
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: pypa/cibuildwheel@v2.23
+        env:
+          CIBW_PLATFORM: pyodide
+          SKLEARN_SKIP_OPENMP_TEST: "true"
+          SKLEARN_SKIP_NETWORK_TESTS: 1
+          CIBW_TEST_REQUIRES: "pytest pandas"
+          CIBW_TEST_COMMAND: "python -m pytest -svra --pyargs sklearn --durations 20 --showlocals"
+
+      - name: Upload wheel artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: cp312-pyodide_wasm32
```

Comment:
```suggestion
          name: pyodide_wheel
```

i.e., this is a bit nitpicky, but I think we should rename it to something more generic, as cp312 won't be valid once we release Pyodide 0.28, and there's just one artifact that's being moved around anyway.